### ISSUE TITLE
Failure to Start - Fedora 41 Silverblue - Flatpak

### ISSUE THREAD

**Issue by KyleRConway** [input-leap/input-leap#2079]
### What happened?

Installed via Flatpak (InputLeap_3.0.2_linux_x86_64.flatpak) on Fedora 41 Silverblue immutable OS. The application would start the GUI but never connect. 

Running via cli `flatpak run io.github.input_leap.InputLeap` included the following error `QDBusTrayIcon encountered a D-Bus error`.

That lead me to this bug report which _may_ be related: https://www.mail-archive.com/kdepim-bugs@kde.org/msg142922.html with the following possible fix? https://invent.kde.org/pim/kleopatra/-/commit/96eccd7c47069fa629a09fe5838b30bf11fceb45#411b2b4f2845fc198ab7e7de06b06d135843fd19

At some point in one of those chains was the following possible workaround (which did not work): 

`export QT_QPA_PLATFORM=XCB && flatpak run io.github.input_leap.InputLeap`

The output is provided below if helpful.

```
qt.qpa.xcb: could not connect to display 
qt.qpa.plugin: From 6.5.0, xcb-cursor0 or libxcb-cursor0 is needed to load the Qt xcb platform plugin.
qt.qpa.plugin: Could not load the Qt platform plugin "xcb" in "" even though it was found.
This application failed to start because no Qt platform plugin could be initialized. Reinstalling the application may fix this problem.

Available platform plugins are: eglfs, linuxfb, minimal, minimalegl, offscreen, vkkhrdisplay, vnc, wayland-egl, wayland, xcb.
```

Not sure if this is a Fedora 41 thing, a Wayland thing, a Flatpak thing, an immutable OS thing, or a combo. Providing what I tracked down here in the hopes it may help in resolution.

Thanks much for this great tool.

### Version

v3.0.2

### If applicable, where did you install Input Leap from?

Flatpak downloaded from Github.

### What OSes are you seeing the problem on? (Check all that apply)

Linux

### If applicable, are you using Wayland or X11?

Wayland

### What OS versions are you using?

Fedora 41 Silverblue

### Any other information

See above. Happy to run a log if you can direct me in obtaining one.

**Comment by ccoenen** [comment]
I am also on Fedora 41 Silverblue and input-leap does not work for me either. I have input-leap `3.0.2-release-release-00000000` (yes, seriously) from the rpm packages installed. I can start it, but when I click "configure server" it will crash.

If I start it from the terminal, I get this output:

```
$ input-leap
System tray is unavailable.
*** WARNING *** The program 'input-leap' called 'DNSServiceRegister()' which is not supported (or only supported partially) in the Apple Bonjour compatibility layer of Avahi.
*** WARNING *** Please fix your application to use the native API of Avahi!
*** WARNING *** For more information see <http://0pointer.de/blog/projects/avahi-compat.html>
QDBusTrayIcon encountered a D-Bus error: QDBusError("org.freedesktop.DBus.Error.ServiceUnknown", "The name is not activatable")
```

All this output happens without any kind of interaction. There is no further output before/after crashing.

My computer running "regular" Fedora 41 including GNOME 47 and wayland does _not_ exhibit the same problem. And since I run this off of rpm packages, I believe the only remaining variable is the immutable system.

**Comment by leifmadsen** [comment]
> If I start it from the terminal, I get this output:
> 
> ```
> $ input-leap
> System tray is unavailable.
> *** WARNING *** The program 'input-leap' called 'DNSServiceRegister()' which is not supported (or only supported partially) in the Apple Bonjour compatibility layer of Avahi.
> *** WARNING *** Please fix your application to use the native API of Avahi!
> *** WARNING *** For more information see <http://0pointer.de/blog/projects/avahi-compat.html>
> QDBusTrayIcon encountered a D-Bus error: QDBusError("org.freedesktop.DBus.Error.ServiceUnknown", "The name is not activatable")
> ```

Same issue on Fedora 41 Desktop running hyprland or sway when clicking on the Configure Server button.

**Comment by shymega** [comment]
@ccoenen Which DM are you using with this error? If it's wlroots-based (Hyprland, sway), then this is an expected error, and there is no workaround at this time.

You will need libportal, xdg-desktop-portal, and xdg-desktop-portal-gnome for Input Leap to work.

**Comment by shymega** [comment]
To all: the original post was specifically about Flatpaks, and the latter two reports are not related to the OP.

**Comment by shymega** [comment]
@KyleRConway Which DE are you using on your installation?

**Comment by KyleRConway** [comment]
> @KyleRConway Which DE are you using on your installation?

Fedora 41 Silverblue immutable OS: https://fedoraproject.org/atomic-desktops/silverblue/

**Comment by shymega** [comment]
> > @KyleRConway Which DE are you using on your installation?
> 
> Fedora 41 Silverblue immutable OS: https://fedoraproject.org/atomic-desktops/silverblue/

@KyleRConway This doesn't tell me the DE/WM. Are you using GNOME? KDE? Sway? And in Wayland or X11?

**Comment by KyleRConway** [comment]
> > > @KyleRConway Which DE are you using on your installation?
> > 
> > 
> > Fedora 41 Silverblue immutable OS: https://fedoraproject.org/atomic-desktops/silverblue/
> 
> @KyleRConway This doesn't tell me the DE/WM. Are you using GNOME? KDE? Sway? And in Wayland or X11?

Apologies. Silverblue is the Gnome variant, and it's running Wayland.

**Comment by shymega** [comment]
@KyleRConway I am unsure if this is an Input Leap bug or Flatpak or Qt.

cc: @p12tic - possibly due to Flatpak sandboxing?

**Comment by leifmadsen** [comment]
All Fedora from 41 and forward are Wayland only, fyi.

**Comment by shymega** [comment]
> All Fedora from 41 and forward are Wayland only, fyi.

Not necessarily. People may modify their Fedora install, and we cannot assume that all Fedora installs will be Wayland. Same for any distro.

**Comment by leifmadsen** [comment]
> > All Fedora from 41 and forward are Wayland only, fyi.
> 
> Not necessarily. People may modify their Fedora install, and we cannot assume that all Fedora installs will be Wayland. Same for any distro.

OK, but Fedora doesn't ship X11 other than in spins any more, so Fedora 41 desktop users will be using Wayland, unless they've done some magic to hack around that.

**Comment by ccoenen** [comment]
This ticket is specifically about flatpak, not `dnf`.

The version in the packages is 3.0.2 right now. On my computer (and also with dnf) I had minor problems with it. 3.0.3 (can be downloaded here, is not yet in the official packages) seems to resolve those for me.

**Comment by shymega** [comment]
@ccoenen Is this now resolved for you with Flatpaks, then? I guess we could close this.

**Comment by ccoenen** [comment]
> @ccoenen [...] Flatpaks [...]

I couldn't tell you because [I](https://darmstadt.social/@claudius/113529155260960728) [started](https://darmstadt.social/@claudius/114617972290030933) to [avoid](https://darmstadt.social/@claudius/114684226030239577) flatpak like the plague. Fedora 43 system packages (rpms) are now at 3.0.3 and work fine.

Maybe @KyleRConway can chime in if he still sees this issue on his end?

**Comment by Medullitus** [comment]
> > [@ccoenen](https://github.com/ccoenen) [...] Flatpaks [...]
> 
> I couldn't tell you because [I](https://darmstadt.social/@claudius/113529155260960728) [started](https://darmstadt.social/@claudius/114617972290030933) to [avoid](https://darmstadt.social/@claudius/114684226030239577) flatpak like the plague. Fedora 43 system packages (rpms) are now at 3.0.3 and work fine.
> 
> Maybe [@KyleRConway](https://github.com/KyleRConway) can chime in if he still sees this issue on his end?

What'z wrong with flatpak? Just curious... Isn't it the safest solution?

**Comment by ccoenen** [comment]
> What'z wrong with flatpak? Just curious... Isn't it the safest solution?

The flipside is, that a lot of stuff plain out does not work. I've had problems anywhere from "drag and drop does not work" to "weird rendering issues" to "system libraries are not available within the flatpak'd app" to "some flatpaks don't start if my NAS is not connected" to "programs that depend on each other can't see each other".

Some of this is flatpak working as designed. Some of this is "community-members put their favorite software onto flatpak while not taking enough care to test it".

Here is software that I have tried on Flatpak first, which have had issues that were *immediately* resolved when installed as RPM: KeePassXC, LibreOffice, Nextcloud client, Rider (IDE), Flameshot, codec issues with VLC (this one is in part on fedora, I know). Kdenlive. I am probably forgetting a few, too.

Yes, I know flatseal. Yes, some issues can be resolved that way. Not all of them. Not by a long shot.

Feel free to mark this as off-topic to the discussion. Sorry about hijacking the issue.

**Comment by shymega** [comment]
Locking this issue as off-topic. It's probably going to get heated. If the issue is persistent, please open a new issue and reference this one.
